File: output/llama_4_maverick/avatar/direct_translation/java/python/codeforces_299_A.py

import sys
n = int(sys.stdin.readline())
a = list(map(int, sys.stdin.readline().split()))
a.sort()
min_val = a[0]
for value in a:
    if value % min_val != 0:
        print(-1)
        sys.exit()
print(min_val)

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/successful_translation/codeforces_299_A.java
import java.util.*;
import java.io.*;

public class codeforces_299_A {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        br.readLine();
        String[] input = br.readLine().split(" ");
        int[] a = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            a[i] = Integer.parseInt(input[i]);
        }
        Arrays.sort(a);
        boolean hasRemainder = false;
        for (int x : a) {
            if (x % a[0] != 0) {
                hasRemainder = true;
                break;
            }
        }
        System.out.println(hasRemainder ? -1 : a[0]);
    }
}
